sklearn/model_selection/_search.py

```diff
@@ -552,59 +593,140 @@ def _fit(self, X, y, labels, parameter_iterable):
             pre_dispatch=pre_dispatch
         )(delayed(_fit_and_score)(clone(base_estimator), X, y, self.scorer_,
                                   train, test, self.verbose, parameters,
-                                  self.fit_params, return_parameters=True,
+                                  self.fit_params, return_parameters=False,
                                   error_score=self.error_score)
           for parameters in parameter_iterable
           for train, test in cv.split(X, y, labels))
 
+        out = np.array(out, dtype=np.float64)
+
         # Out is a list of triplet: score, estimator, n_test_samples
-        n_fits = len(out)
+        self.n_fits_ = out.shape[0]
+
+        self.n_candidates_ = self.n_fits_ // n_splits
+        self.n_parameters_ = len(self._parameter_names)
+
+        res_shape = (self.n_candidates_,)
+
+        # Lets not initite this everytime and reuse the same array.
+        test_scores = np.empty((n_splits,), dtype=np.float64)
+        test_sample_counts = np.empty((n_splits,), dtype=np.float64)
+
+        search_results = dict()
+
+        for param in self._parameter_names:
+            # One column to record the values of each parameter
+            search_results["param_%s" % param] = (
+                np.ma.masked_all(res_shape, dtype=object))
+
+        # Make a column for each split of each metric
+        for split_i in range(n_splits):
+            search_results["test_split_%s" % split_i] = (
+                np.empty(res_shape, dtype=np.float64))
+        search_results["test_mean"] = np.empty(res_shape, dtype=np.float64)
+        search_results["test_std"] = np.empty(res_shape, dtype=np.float64)
+        search_results["test_rank"] = np.empty(res_shape, dtype=int)
+
+        for candidate_i in range(self.n_candidates_):
+            grid_start = candidate_i * n_splits
+            grid_end = grid_start + n_splits
+
+            test_scores = out[grid_start:grid_end, 0]
```

Comment:
I fail to understand how this could be done with a reshape. I understand we could `zip(*out)` and get lists which could be converted to numpy arrays.

But again they should still be sliced (`grid_start:grid_end`) to get the block of scores/n_test_samples for that candidate right?